### PR TITLE
Replicate files

### PR DESCRIPTION
**Pull request by tristan-cognite**
- add file replication methods
- generalize common file/event/timeseries replication operations more?

### CODE REVIEW

File: cognite/replicator/files.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        src_file: The file from the source to be replicated to destination.
+        src_dst_ids_assets: A dictionary of all the mappings of source asset id to destination asset id.
+        project_src: The name of the project the object is being replicated from.
+        runtime: The timestamp to be used in the new replicated metadata.
+
+    Returns:
+        The replicated file to be created in the destination.
+    """
+    logging.debug(f"Creating a new file based on source file id {src_file.id}")
+    mime_type = mimetypes.types_map.get(f".{src_file.mime_type}", src_file.mime_type)
```
Comment (eventh) [inline cognite/replicator/files.py]: Why use format string here? Can't you use src_file.mime_type directly, or if you need it be string, use the str() method. :)

File: cognite/replicator/files.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        dst_file: The file from the destination that needs to be updated to reflect changes made to its source file.
+        src_dst_ids_assets: A dictionary of all the mappings of source asset id to destination asset id.
+        project_src: The name of the project the object is being replicated from.
+        runtime: The timestamp to be used in the new replicated metadata.
+
+    Returns:
+        The updated file object for the replication destination.
+    """
+    logging.debug(f"Updating existing file {dst_file.id} based on source file id {src_file.id}")
+
```
Comment (eventh) [inline cognite/replicator/files.py]: Is mime_type immutable?

Comment (tristan-cognite) [reply cognite/replicator/files.py]: Per https://docs.cognite.com/api/v1/#operation/updateFiles it looks like that is the case

File: cognite/replicator/files.py (hunk 1 above)
Comment (tristan-cognite) [reply cognite/replicator/files.py]: We store "pdf" but mimetypes library needs ".pdf" - maybe str concat "." + str(...) would be more clear

Comment (eventh) [reply cognite/replicator/files.py]: Ah, I missed the dot, then it is perfect :)

File: cognite/replicator/files.py (hunk 2 above)
Comment (eventh) [reply cognite/replicator/files.py]: Interesting :)

**Pull request merged**